language/bytecode-verifier/src/verifier.rs

```diff
@@ -58,18 +58,9 @@ fn verify_main_signature_impl(script: &CompiledScript) -> PartialVMResult<()> {
         match arg_type {
             S::Bool | S::U8 | S::U64 | S::U128 | S::Address => true,
             S::Vector(inner) => match &**inner {
-                S::U8 => true,
-                S::Bool
-                | S::U64
-                | S::U128
-                | S::Address
-                | S::Signer
-                | S::Struct(_)
-                | S::Vector(_)
-                | S::StructInstantiation(_, _)
-                | S::Reference(_)
-                | S::MutableReference(_)
-                | S::TypeParameter(_) => false,
+                S::Bool | S::U8 | S::U64 | S::U128 | S::Address => true,
+                S::Vector(vec_type) => is_valid_arg_type(idx, vec_type),
```

Comment:
Would this accept `vector<&signer>`, `vector<vector<&signer>>`, etc  (which we don't want)? Looks like it might, but I can't tell... Might  be worth a test.

Reply:
absolutely correct! nice!

Reply:
Hmm isn't the script already verified at the moment? `vector<&...>` isn't really a structurally valid type.

Reply:
yes true! and I read what @sblackshear said as vector<signer> which would have been a problem.
So ignore the reference and that is what happened ;)

Reply:
though maybe that was ok too.... I'll look again as I am trying to reconcile all the functions related to constants